src/test/java/com/browserstack/BrowserStackTest.java

```diff
@@ -1,89 +1,30 @@
-import org.testng.annotations.Parameters;
 
 import com.codeborne.selenide.WebDriverRunner;
 
 
 public class BrowserStackTest {
-	public RemoteWebDriver driver;
-	private Local l;
-
-	public static String username;
-	public static String accessKey;
-	public static String sessionId;
-
-	@BeforeMethod(alwaysRun=true)
-	@Parameters(value={"config", "environment"})
-	public void setUp(String config_file, String environment) throws Exception {
-		JSONParser parser = new JSONParser();
-		JSONObject config = (JSONObject) parser.parse(new FileReader("src/test/resources/conf/" + config_file));
-		JSONObject envs = (JSONObject) config.get("environments");
-
-		DesiredCapabilities capabilities = new DesiredCapabilities();
-
-		Map<String, String> envCapabilities = (Map<String, String>) envs.get(environment);
-		Iterator it = envCapabilities.entrySet().iterator();
-    	while (it.hasNext()) {
-      		Map.Entry pair = (Map.Entry)it.next();
-      		capabilities.setCapability(pair.getKey().toString(), pair.getValue().toString());
-    	}
-
-		Map<String, String> commonCapabilities= (Map<String, String>) config.get("capabilities");
-		it = commonCapabilities.entrySet().iterator();
-		while (it.hasNext()) {
-			Map.Entry pair = (Map.Entry)it.next();
-			if (capabilities.getCapability(pair.getKey().toString()) == null) {
-				capabilities.setCapability(pair.getKey().toString(), pair.getValue().toString());
-			}
-		}
-
-		username = System.getenv("BROWSERSTACK_USERNAME");
-		if (username == null) {
-			username = (String) config.get("user");
-		}
-
-		accessKey = System.getenv("BROWSERSTACK_ACCESS_KEY");
-		if (accessKey == null) {
-			accessKey = (String) config.get("key");
-		}
-
-		if (capabilities.getCapability("browserstack.local") != null && capabilities.getCapability("browserstack.local") == "true") {
-			l = new Local();
-			Map<String, String> options = new HashMap<String, String>();
-			options.put("key", accessKey);
-			String currentTime = String.valueOf(System.currentTimeMillis());
-			options.put("localIdentifier", currentTime);
-			capabilities.setCapability("browserstack.localIdentifier", currentTime);
-			l.start(options);
-		}
-
-		driver = new RemoteWebDriver(new URL("http://" + username + ":" + accessKey + "@" + config.get("server") + "/wd/hub"), capabilities);
-		driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
-		sessionId = driver.getSessionId().toString();
-
-		WebDriverRunner.setWebDriver(driver);
-	}
-
-	@AfterMethod(alwaysRun=true)
-	public void tearDown() throws Exception {
-		WebDriverRunner.closeWebDriver();
-		if (l != null) l.stop();
-	}
+    public RemoteWebDriver driver;
+
+    @BeforeMethod(alwaysRun = true)
+    public void setUp() throws Exception {
+			MutableCapabilities capabilities = new MutableCapabilities();
+			driver = new RemoteWebDriver(new URL("https://hub-cloud.browserstack.com/wd/hub"), capabilities);
+			driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
```

Comment:
implicit wait is deprecated in newer versions.

Reply:
This works in selenium 4 too. Webdriverwait is introduced for until and other functions. 
https://www.seleniumeasy.com/selenium-tutorials/waits-and-timeout-selenium-4